psydac/core/kernels.py

```diff
@@ -391,7 +397,7 @@ def eval_fields_3d_weighted(nc1: int, nc2: int, nc3: int, f_p1: int, f_p2: int,
                                                                 span_2 - f_p2:1 + span_2,
                                                                 span_3 - f_p3:1 + span_3]
 
-                arr_fields[:, :, :, :] = 0.0
+                arr_fields[:, :, :, :] *= 0.0
```

Comment:
Was the original code not working in the complex case? That would sound very strange.

Reply:
I thought that it was not working but I test and it seems to work. What do you think is the most efficient way to do it ?

Reply:
Oh I just found the issue, in fact I can use `=0.0` for a vector but not for a scalar. Thus, I have made the change for to much variable but the issue is still relevant in the case of a complex scalar with `scalar=0.0`.

Reply:
In this commit, I removed all the `*=` when the lhs is an array
[a63c4f1](https://github.com/pyccel/psydac/pull/312/commits/a63c4f1336baa69f1ecd7ea0477a70a69b7c55fa)